scripts/TestGen.hx

```diff
@@ -0,0 +1,380 @@
+		Sys.println('Creating test/Test.hx');
+		File.saveContent('${dir}test/Test.hx', testTmpl.execute(
+			{
+				stories: stories.join("\n\t\t")
+			}));
+
+		// copy test.hxml from hello-world
+		Sys.println('Copying test.hxml');
+		File.copy('${exercisesDir}hello-world/test.hxml', '${dir}test.hxml');
+
+		Sys.println("Generation Successful");
+		Sys.println("Don't forget to update config.json!");
+	}
+
+	private static function generateTests(cases:Array<Dynamic>, methods:Map<String, {order:Int, method:String}>, stories:Array<String>,
+			testCases:Map<String, Array<String>>, meta:Array<String>, ?story:String)
+	{
+		for (testCase in cases)
+		{
+			if (testCase.cases == null)
+			{
+				// if story does not exist make a new one
+				if (story == null)
+				{
+					story = toUpperCamel(exercise, " ");
+				}
+				if (story != null && !testCases.exists(story))
+				{
+					stories.push(story);
+					testCases.set(story, new Array<String>());
+				}
+
+				// add description to .meta/test.toml
+				meta.push('# ${testCase.description}\n');
+				// add uuid in double quotes = true to .meta/test.toml
+				meta.push('"${testCase.uuid}" = true\n\n');
+
+				var method = testCase.property;
+
+				// iterate over inputs and add them to array to join later
+				var inputs = new Array<Dynamic>();
+				var types = new Array<String>();
+
+				for (obj in Reflect.fields(testCase.input))
+				{
+					var value = processField(Reflect.field(testCase.input, obj));
+					inputs.push(value.obj);
+
+					// need the types of inputs for newly found methods
+					if (!methods.exists(method))
+					{
+						types.push('${obj}: ${value.type}');
+					}
+				}
+
+				var expected = processField(testCase.expected);
+
+				// Check if the method exits otherwise execute template
+				// need to check if the type is a Generic Array and warn if this happens
+				// This warning may correct itself if the story has multiple cases
+				// and the first case happens check against an empty array.
+				if (!methods.exists(method) && expected.type != "Array")
+				{
+					var count = 0;
+					for (method in methods.keys())
+						count++;
+					methods.set(method,
+						{
+							order: count,
+							method: methodsTmpl.execute(
+								{
+									method: method,
+									args: types.join(", "),
+									returnType: expected.type
+								})
+						});
+				} else if (expected.type == "Array")
+				{
+					Sys.println("WARNING: Unidentifiable generic found for method: " + method);
+					Sys.println("Please verify that it was corrected!");
+				}
+
+				var shouldAssert:String;
+				if (expected.type.startsWith("Array"))
+				{
+					shouldAssert = "containExactly";
+				} else
+				{
+					shouldAssert = "be";
+				}
+
+				var testCriteria = '${toUpperCamel(exercise)}.${method}(${inputs.join(", ")}).should.${shouldAssert}(${expected.obj});';
+				if (testCases.get(story).length != 0)
+				{
+					testCriteria = 'pending("Skipping");\n\t\t\t\t' + testCriteria;
```

Comment:
We don't use pending("Skipping") but xit instead of it to mark a test as pending. Every test after the first one globally should be pending even if it's in a different describe block